scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -548,21 +549,6 @@ def handle(self):
                 self.make_device_available(serial_device)
             return
 
-        ser.flush()
```

Comment:
there are scenarios that below code works to clean a uart driver buffer at host side, so maybe add an option to disable below code, and leave it as default enabled will be more safer.

Reply:
@golowanow , the old code in introduced by @smrtos , and I see the prupose of code is
'''
The logic to drain the leftover logs of previous test
'''
so if a former case keeps sending data, the reset_input_buffer() may not enough, @smrtos can you clarify this?

Reply:
>  if a former case keeps sending data, the reset_input_buffer() may not enough

@hakehuang, if a former case keeps running and sending data then neither `reset_input_buffer()`, nor some hardcoded timeout/discard buffer size can help to stop it, but `serial_pty` or `pre_script`, isn't it ?

Reply:
@golowanow, I agree with you, let's see what happen after merge your changes, I did see any issue from NXP board sige.